flow/layers/image_filter_layer.cc

```diff
@@ -39,12 +40,34 @@ void ImageFilterLayer::Diff(DiffContext* context, const Layer* old_layer) {
 void ImageFilterLayer::Preroll(PrerollContext* context,
                                const SkMatrix& matrix) {
   TRACE_EVENT0("flutter", "ImageFilterLayer::Preroll");
+
   Layer::AutoPrerollSaveLayerState save =
       Layer::AutoPrerollSaveLayerState::Create(context);
 
+  context->raster_cached_entries.emplace_back(
+      RasterCacheEntry(this, *context, matrix));
+  auto current_index = context->raster_cached_entries.size();
+
+  auto& cache_entry = context->raster_cached_entries.back();
+
   SkRect child_bounds = SkRect::MakeEmpty();
   PrerollChildren(context, matrix, &child_bounds);
 
+  cache_entry.num_child_entries =
+      context->raster_cached_entries.size() - current_index;
+  auto cache_type = NeedCaching(context, matrix);
+
+  cache_entry.need_caching = false;
+  if (cache_type == CacheableLayer::CacheType::kCurrent) {
+    cache_entry.layer = this;
+  } else if (cache_type == CacheableLayer::CacheType::kChildren) {
+    // Replace Cacheable child
+    auto* cacheable_child = GetCacheableChild();
+    if (cacheable_child) {
+      cache_entry.UpdateRasterCacheEntry(cacheable_child, *context, matrix);
```

Comment:
Only the layer pointer needs to be updated here. The context and the matrix are already set correctly at the top of this method.

Reply:
Also, the need_caching flag should be set to true here.